pkg/operator2/idp.go

```diff
@@ -292,6 +313,77 @@ func (c *authOperator) transportForCARef(ca configv1.ConfigMapNameReference, key
 	return transportFor("", caData, nil, nil)
 }
 
+func (c *authOperator) checkOIDCPasswordGrantFlow(
+	tokenURL, clientID string,
+	caRererence configv1.ConfigMapNameReference,
+	clientSecretReference configv1.SecretNameReference,
+) (bool, error) {
+	secret, err := c.secrets.Secrets("openshift-config").Get(clientSecretReference.Name, metav1.GetOptions{})
+	if err != nil {
+		return false, fmt.Errorf("couldn't get the referenced secret: %v", err)
+	}
+
+	// check whether we already attempted this not to send unneccessary login
+	// requests against the provider
+	if cachedResult, ok := oidcPasswordChecks[secret.ResourceVersion]; ok {
+		klog.V(4).Info("using cached result for OIDC password grant check")
+		return cachedResult, nil
+	}
+
+	clientSecret, ok := secret.Data["clientSecret"]
+	if !ok || len(clientSecret) == 0 {
+		return false, fmt.Errorf("the referenced secret does not contain the 'clientSecret' key")
+	}
+
+	transport, err := c.transportForCARef(caRererence, "ca")
+	if err != nil {
+		return false, fmt.Errorf("couldn't get a transport for the referenced CA: %v", err)
+	}
+
+	// prepare the grant-checking query, "openid" scope should be enough
+	query := url.Values{}
+	query.Add("client_id", clientID)
+	query.Add("client_secret", string(clientSecret))
+	query.Add("grant_type", "password")
+	query.Add("scope", "openid")
+	query.Add("username", "test")
+	query.Add("password", "test")
+	body := strings.NewReader(query.Encode())
+
+	// prepare the request, require application/json for simple parsing the response
+	req, err := http.NewRequest("POST", tokenURL, body)
+	if err != nil {
+		return false, err
+	}
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	req.Header.Set("Accept", "application/json")
+
+	reqBytez, err := httputil.DumpRequest(req, true)
+	klog.V(1).Infof("teh bytes be: %s", reqBytez)
```

Comment:
nit: s/teh/the

Reply:
I forgot to remove this, it was for testing purposes. I tend to amuse myself by misspelling "the" this way

Reply:
Btw - this would log the client secret so we definitely do not want that